### PR TITLE
Implement HTML Editor for Candidate Job Experience notes

### PR DESCRIPTION
**Pull request by timhill-iress**
Covers:
Candidate entry screen in Candidate Portal
Candidate screen in Admin Portal
CV in Public Portal
CV Download PDF in Admin Portal

Note: ElasticSearch data requires migration!

### CODE REVIEW

**Comment by timhill-iress** [comment]
PR set as draft, as not ready for merging yet. I've just created the PR to share with others.

I still need to document the process for updating ElasticSearch index - and understand more about the TBB ElasticSearch setup.
Also discuss what other text entry needs to allow HTML

File: docker-compose.yaml
Comment (camerojo) [inline docker-compose.yaml:1]: I don't know why this docker config stuff is needed Tim.

We currently don't use Docker to create the image. Instead we used a tool from Google called jib which is only ever used in the GitHub build.

See https://cloud.google.com/java/getting-started/jib

File: ui/candidate-portal/src/app/components/common/candidate-job-experience-form/candidate-job-experience-form.component.html
Comment (camerojo) [inline ui/candidate-portal/src/app/components/common/candidate-job-experience-form/candidate-job-experience-form.component.html:1]: Any reason for leaving out the class="form-control"?

**Review by camerojo — COMMENTED** [review]
All looks good to me Tim.

Just a couple of comments/questions

**Comment by timhill-iress** [comment]
Hi @sadatmalik, 
Should we just close this PR?
https://github.com/Talent-Catalog/talentcatalog/pull/57 implemented this feature, but in a different way. It extracted the text rather than using Elastic Search to filter the HTML tags out.
I know you were going to do some work on Elastic Search, and we thought we might revisit this approach then. The changes would be different though as part of this is already in the code base.

What do you think?

Tim

**Comment by camerojo** [comment]
@sadatmalik - As @timhill-iress suggests, I think we should close this PR. 

I have verified that what we currently have in production is, in fact, stripping all HTML from text using a JSoup based implementation of Tim's TextExtractor interface (JsoupTextExtracterImpl.java). In particular, see the copy method of CandidateEs.

**Comment by sadatmalik** [comment]
Agreed @timhill-iress and @camerojo - closing this PR.

**Pull request closed without merging**